src/lss.jl

```diff
@@ -216,10 +218,99 @@ function stationary_distributions(lss::LSS; max_iter=200, tol=1e-5)
     return mu_x, mu_y, Sigma_x, Sigma_y
 end
 
-
 function geometric_sums(lss::LSS, bet, x_t)
+    is_stable(lss)
     I = eye(lss.n)
     S_x = (I - bet .* lss.A) \ x_t
     S_y = lss.G * S_x
     return S_x, S_y
 end
+
+doc"""
+Test for stability of linear state space system.
+First removes the constant row and column.
+
+#### Arguments
+
+- `lss::LSS` The linear state space system
+
+#### Returns
+
+- `stable::Bool` Whether or not the system is stable
+
+"""
+function is_stable(lss::LSS)
+
+    # Get version of A without constant row/column
+    A = remove_constants(lss)
+
+    # Check for stability
+    stable = is_stable(A)
+    if stable==false
+        warn("Stationary distribution does not exist.")
+    end
+    return stable
+
+end
+
+doc"""
+General function for testing for stability of matrix ``A``. Just
+checks that eigenvalues are less than 1 in absolute value.
+
+#### Arguments
+
+- `A::Matrix` The matrix we want to check
+
+#### Returns
+
+- `stable::Bool` Whether or not the matrix is stable
+
+"""
+function is_stable(A::AbstractMatrix)
+
+    # Check for stability by testing that eigenvalues are less than 1
+    stable = true
+    d = eigvals(A)
+    if maximum(abs, d) > 1.0
+        stable = false
+    end
+    return stable
+
+end
+
+doc"""
+Finds the row and column, if any,  that correspond to the constant 
+term in a `LSS` system and removes them to get the matrix that needs 
+to be checked for stability.
+
+#### Arguments
+
+- `lss::LSS` The linear state space system
+
+#### Returns
+
+- `A::Matrix` The matrix A with constant row and column removed
+
+"""
+function remove_constants(lss::LSS)
```

Comment:
I think we can do better than this:

I think the code below is a pretty slick way to do it but am open to feedback -- It is a little shorter and should use a little less memory (this is probably very minor improvements but if we can get away with a little less array creation then it doesn't hurt).

```julia
function remove_constants(A::LSS)
    # Get size of matrix
    A = lss.A
    n, m = size(A)
    @assert n==m

    # Sum the absolute values of each row -> Do this because we
    # want to find rows that the sum of the absolute values is 1
    row_sums_to_one = (vec(sum(abs, A, 2) - 1.0)) .< 1e-14
    is_ii_one = map(i->abs(A[i, i] - 1.0) < 1e-14, 1:n)
    not_constant_index = .!(row_sums_to_one .& is_ii_one)

    return A[not_constant_index, not_constant_index]
end
```

Reply:
Done.